packages/integration_test/lib/integration_test_driver.dart

```diff
@@ -30,9 +30,9 @@ typedef ResponseDataCallback = FutureOr<void> Function(Map<String, dynamic>);
 Future<void> writeResponseData(
   Map<String, dynamic> data, {
   String testOutputFilename = 'integration_response_data',
-  String destinationDirectory,
+  String? destinationDirectory,
 }) async {
-  assert(testOutputFilename != null);
+  assert(testOutputFilename != null); // ignore: unnecessary_null_comparison
```

Comment:
This line seems no longer necessary when we have null-safety.

Reply:
Same comment as above - these are still helpful when working in mixed/unsound mode.